apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -182,6 +183,24 @@ export class ProjectChangeAnalyzer {
    */
```

Comment:
My (admittedly very vague) understanding of the problem is that we have two fundamentally different cases:

1. `rush change` cares only about whether a PR diff touches files under a project folder
2. The incremental build is trying to determine which projects are semantically "impacted" by a diff, which includes shrinkwrap changes and whatever that `forIncrementalBuild` logic is doing.

It's questionable whether the implementation of those two things should be mixed together internally, although I'm okay with keeping it that way for now.

But the `public` API should be pretty explicit about this difference, so we don't get confused again.  Like maybe they should even have different verbs, something like:

1. `getProjectsWithChangesAsync()` has docs saying it is for `rush change` and does not expose any `ignoreShrinkwrapChanges` (why would that ever be relevant to `rush change`?)
2. `getProjectsImpactedByDiffAsync()` has docs saying it's for `--changed-since` and considers `rush-project.json` settings

Reply:
Do you have an example of a change that merits republishing a project but does not merit rebuilding the project?